core/src/main/java/org/apache/calcite/avatica/AvaticaUtils.java

```diff
@@ -211,6 +211,7 @@ public static <T> T instantiatePlugin(Class<T> pluginClass,
       String className) {
     String right = null;
     String left = null;
+    Object value = null;
```

Comment:
Does`AvaticaUtils#instantiatePlugin` and `ConnectionConfigImpl#getPlugin`'s java doc need to explicitly say that the 'INSTANCE' could be a `ThreadLocal`?

Reply:
In fa3244a99 I have improved the javadoc for the 'Class#FOO' case.

I don't think we should allow `INSTANCE` to be a `ThreadLocal` because in the singleton pattern INSTANCE should have the same type as its enclosing class.